pkg/server/debug/pprofui/server.go

```diff
@@ -97,6 +97,162 @@ func (s *Server) parsePath(reqPath string) (profType string, id string, remainin
 	}
 }
 
+// validateProfileRequest validates that the request does not contian any
+// conflicting or invalid configurations.
+func validateProfileRequest(req *serverpb.ProfileRequest) error {
+	switch req.Type {
+	case serverpb.ProfileRequest_GOROUTINE:
+	case serverpb.ProfileRequest_CPU:
+		if req.LabelFilter != "" {
+			return errors.Newf("filtering using a pprof label is unsupported for %s", req.Type.String())
+		}
+	default:
+		if req.NodeId == "all" {
+			return errors.Newf("cluster-wide collection is unsupported for %s", req.Type.String())
+		}
+
+		if req.WithLabels {
+			return errors.Newf("profiling with labels is unsupported for %s", req.Type.String())
+		}
+
+		if req.LabelFilter != "" {
+			return errors.Newf("filtering using a pprof label is unsupported for %s", req.Type.String())
+		}
+	}
+
+	return nil
+}
+
+func constructProfileReq(r *http.Request, profileName string) (*serverpb.ProfileRequest, error) {
```

Comment:
Is this code movement? For future PRs, see if there is a way to have a separate commit that mechanically moves code (and the reviewer can trust that's all that's happening) and then layer additional modifications of the logic on top in separate commits. As is, I basically have to review from scratch, which is a lot more work.

Reply:
This was just code movement. Yeah I realize that a separate refactor only commit would have made this a lot easier. The truth is I didn't realize this change would balloon into what it did but I should've made the call to switch to smaller commits before getting in so deep. Noted for next time!